src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -426,8 +456,9 @@ public static ulong GetKey(ReadOnlySpan<byte> propertyName)
                 key = 0;
             }
 
-            // Embed the propertyName length in the last two bytes.
-            key |= (ulong)propertyName.Length << 48;
+            // Embed the length in the last byte.
+            key |= (ulong)Math.Min(propertyName.Length, 0xFF) << 0x38;
```

Comment:
nit: I prefer using decimal values for the bit-shifting, rather than hex. Consider reverting. To me, that's easier to reason about. What was the rational for changing these?

Reply:
I found it easier to verify correctness with hex.

Reply:
I'll update it to use a more readable form of `8 * x` where `x` is the amount of bytes to shift.